polls.php

```diff
@@ -255,16 +255,22 @@
 			$options[$i] = '';
 		}
 
-		if(trim($options[$i]) != "")
+		if($mybb->settings['polloptionlimit'] != 0 && my_strlen($options[$i]) > $mybb->settings['polloptionlimit'])
 		{
-			$optioncount++;
+			$lengtherror = 1;
+			break;
 		}
 
-		if(my_strlen($options[$i]) > $mybb->settings['polloptionlimit'] && $mybb->settings['polloptionlimit'] != 0)
+		if(strpos($options[$i], '||~|~||') !== false)
 		{
-			$lengtherror = 1;
+			$sequenceerror = 1;
```

Comment:
Minor nitpick, but can we rename this to `sequence_error`? IMO it's easier to read that way.